api/agent/agent.go

```diff
@@ -204,6 +206,11 @@ func (a *agent) Submit(callI Call) error {
 	// to make this remove the container asynchronously?
 	defer slot.Close() // notify our slot is free once we're done
 
+	err = a.fireBeforeCall(ctx, call.Model())
```

Comment:
please push this into `call.Start` after the other bits there, to keep `Submit` small and also `Start` may block the call from running, in which case we may not want `BeforeCall` to be called (?)

Reply:
Ok, doing this